apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/NamespaceService.java

```diff
@@ -392,16 +392,20 @@ private boolean isNamespaceNotPublished(Namespace namespace) {
       Item lastItem = itemService.findLastOne(namespaceId);
       return lastItem != null;
     }
-
-    Date lastPublishTime = latestRelease.getDataChangeLastModifiedTime();
-    List<Item> itemsModifiedAfterLastPublish = itemService.findItemsModifiedAfterDate(namespaceId, lastPublishTime);
-
-    if (CollectionUtils.isEmpty(itemsModifiedAfterLastPublish)) {
-      return false;
-    }
-
+    
+    // Fixed by shuangrui at 2019/9/5
     Map<String, String> publishedConfiguration = gson.fromJson(latestRelease.getConfigurations(), GsonType.CONFIG);
-    for (Item item : itemsModifiedAfterLastPublish) {
+    List<Item> itemsNew = itemService.findItemsKeyNonEmpty(namespaceId);
```

Comment:
How about calling `com.ctrip.framework.apollo.biz.service.ItemService#findItemsWithoutOrdered(java.lang.Long)` and filter those empty items in memory? So that we don't need to add the findByNamespaceIdAndKeyNot method.